src/com/owncloud/android/ui/helpers/FileOperationsHelper.java

```diff
@@ -134,6 +136,37 @@ public void openFile(OCFile file) {
     }
 
     /**
+     * Show dialog to allow the user to choose an app to send the private link of an {@link OCFile},
+     * or copy it to clipboard.
+     *
+     * @param context
+     * @param file @param file {@link OCFile} which will be shared with internal users
+     * @param account
+     */
+    public void copyOrSendPrivateLink(Context context, Account account, OCFile file) {
+
+        try {
+
+            // Parse remoteId
+            String remoteId = file.getRemoteId();
+
+            String parsedRemoteId = remoteId.substring(0, Math.min(remoteId.length(), 8));
+
+            String fileId = Integer.valueOf(parsedRemoteId).toString();
+
+            Uri uri = Uri.parse(com.owncloud.android.lib.common.accounts.
```

Comment:
Yeah, you're right. Parsing the String to Uri is not needed, I can use directly the link obtained in AccountUtils.getUrlForFile to share it later.